File: soccer-league/soccer-league/Data/Mappers/LeagueMapper.swift
import Foundation

struct LeagueMapper {
    static func toDomain(_ dto: LeagueDTO) -> League {
        League(
            id: dto.id,
            name: dto.name,
            imageURL: dto.imageUrl.flatMap { URL(string: $0)?.pandaScoreThumbnail }
        )
    }
}

File: csgo-leagues-android/app/src/main/java/com/voidx/csgo/data/mapper/LeagueMapper.kt
package com.voidx.csgo.data.mapper

import com.voidx.csgo.data.dto.LeagueDTO
import com.voidx.csgo.domain.entity.League

object LeagueMapper {
    fun toDomain(dto: LeagueDTO): League =
        League(
            id = dto.id,
            name = dto.name,
            imageUrl = dto.imageUrl.toPandaScoreThumbnailUrl(),
        )
}
